src/ratbag_emu/protocol/hidpp20.py

```diff
@@ -53,11 +53,9 @@ class HIDPP20Device(BaseDevice):
     '''
     Init routine
     '''
-    def __init__(self, rdesc=None, info=None, feature_table=None,
-                 name='Generic HID++ 2.0 Device', shortname='generic-hipp20'):
-        super().__init__(rdesc, info, name, shortname)
-        self.protocol = 'HID++ 2.0'
-        self.feature_table = feature_table
+    def __init__(self, obj):
+        super().__init__(obj)
+        self.feature_table = obj['hidpp20']['feature_table']
```

Comment:
having a class method `from_json()` as mentioned above would allow you to not modify this file for no reasons.